internal/io/http/rest_sink.go

```diff
@@ -40,7 +42,17 @@ func (ms *RestSink) Validate(props map[string]interface{}) error {
 
 func (ms *RestSink) Configure(ps map[string]interface{}) error {
 	conf.Log.Infof("Initialized rest sink with configurations %#v.", ps)
-	return ms.Validate(ps)
+	err := ms.Validate(ps)
+	if err != nil {
+		return err
+	}
+	if ms.config.Compression != "" {
+		ms.compressor, err = compressor.GetCompressor(ms.config.Compression)
+		if err != nil {
```

Comment:
I guess the compressor.GetCompressor error message is clear enough. Though, no harm to make it more specific.

Reply:
fixed